kikuchipy/signals/ebsd_master_pattern.py

```diff
@@ -289,3 +407,224 @@ class LazyEBSDMasterPattern(EBSDMasterPattern, LazySignal2D):
+    npx : int
+        Number of pixels on the master pattern in the x direction.
+    npy : int
+        Number of pixels on the master pattern in the y direction.
+
+    Returns
+    ----------
+    nii : numpy.ndarray
+        Row coordinate of a point.
+    nij : numpy.ndarray
+        Column coordinate of a point.
+    niip : numpy.ndarray
+        Row coordinate of neighboring point.
+    nijp : numpy.ndarray
+        Column coordinate of a neighboring point.
+    di : numpy.ndarray
+        Interpolation weight factor.
+    dj : numpy.ndarray
+        Interpolation weight factor.
+    dim : numpy.ndarray
+        Interpolation weight factor.
+    djm : numpy.ndarray
+        Interpolation weight factor.
+    """
+    # Normalized direction cosines to Rosca-Lambert projection
+    xy = (
+        scale
+        * LambertProjection.project(rotated_direction_cosines)
+        / (np.sqrt(np.pi / 2))
+    )
+
+    i = xy[..., 0]
+    j = xy[..., 1]
+    nii = i.astype(int) + scale
+    nij = j.astype(int) + scale
+    niip = nii + 1
+    nijp = nij + 1
+    niip = np.where(niip < npx, niip, nii)
+    nijp = np.where(nijp < npy, nijp, nij)
+    nii = np.where(nii < 0, niip, nii)
+    nij = np.where(nij < 0, nijp, nij)
+    di = i - nii + scale
+    dj = j - nij + scale
+    dim = 1.0 - di
+    djm = 1.0 - dj
+
+    return (
+        nii.astype(int),
+        nij.astype(int),
+        niip.astype(int),
+        nijp.astype(int),
+        di,
+        dj,
+        dim,
+        djm,
+    )
+
+
+def _get_patterns_chunk(
+    r: Rotation,
+    dc: Vector3d,
+    master_north: np.ndarray,
+    master_south: np.ndarray,
+    npx: int,
+    npy: int,
+    rescale: bool,
+    dtype_out=np.float32,
+) -> np.ndarray:
+    """
+    Get the EBSD patterns on the detector for each rotation in the chunk.
+
+    Each pattern is found by a bi-quadratic interpolation of the master pattern
+    as described in EMsoft.
+
+    Parameters
+    ----------
+    r : Rotation
+        Rotation object with all the rotations for a given chunk.
+    dc : Vector3d
+        Direction cosines unit vector between detector and sample.
+    master_north : numpy.ndarray
+        Northern hemisphere of the master pattern.
+    master_south : numpy.ndarray
+        Southern hemisphere of the master pattern.
+    npx : int
+        Number of pixels in the x-direction on the master pattern.
+    npy: int
+        Number of pixels in the y-direction on the master pattern.
+    rescale : bool
+        Whether to call rescale_intensities() or not.
+    dtype_out : numpy.dtype
+        Data type of the returned patterns.
+
+    Returns
+    ----------
+    simulated : numpy.ndarray
+        Ndarray with shape (n, y, x) containing all the patterns.
+
+
+    """
+    m = r.shape[0]
+    simulated = np.empty(shape=(m,) + dc.shape, dtype=dtype_out)
+
+    scale_factor = (npx - 1) / 2
+
+    for i in range(m):
+        rot_dc = Rotation(r[i]) * dc
+        (
+            nii,
```

Comment:
Can you get these eight parameters on one line? Like

```python
(
    nii, nij, niip, nijp, di, dj, dim, djm
) = _get_lambert_interpolation_parameters(
    ...
)
```

Reply:
Black doesn't allow for it, even with  (..., djm) and (..., djm,)